### DISCUSSION TITLE
Copilot Extensions now in Limited Public Beta

### DISCUSSION THREAD

**Discussion by goldieowner17** [community/community#125067] [Copilot News and Announcements]
![GitHub Copilot Extensions Limited Beta announcement](https://github.com/github/release-assets/assets/5484339/05f8a107-1e43-4846-91e1-d49cf7f7636c)
GitHub Copilot Extensions is now in [Limited Public Beta](https://aka.ms/githubcopilotextensions). With extensions, you can now extend the capabilities of GitHub Copilot Chat and enhance the experience to perform a wide range of actions across third-party tools, services, and data. Create feature flags, check log errors, access API documentation, and even deploy your application to the cloud all through natural language.

Currently, access to Copilot Extensions is limited to a small set of users and we are planning to gradually roll out larger-scale access as we learn more & gather feedback. To learn more about Copilot Extensions and how to use them, check out the [documentation](https://docs.github.com/copilot/github-copilot-chat/github-copilot-extensions/about-github-copilot-extensions) and the [GitHub Marketplace](https://github.com/marketplace?type=apps&copilot_app=true). 

If you're looking to build a GitHub Copilot Extension, there are several ways. Join the [Copilot Partner Program](https://github.com/features/preview/copilot-partner-program) to explore opportunities to bring your developer tools and services into the GitHub Copilot ecosystem and have extensions light up across all supported clients (VS Code, Visual Studio, and Github.com). If you're looking to build a VS Code extension specifically, documentation is available [here](https://code.visualstudio.com/api/extension-guides/chat).

**Comment by Anequit** [comment]
Will languages offer official extensions, so that copilot has up to date documentation on the language?

**Reply by barbermjgt212** [reply]
QUE GENIAL los idiomas son necesarios para latinoamericanos 
que genial podremos tener mas comunicacion

**Reply by Lakiluktrs** [reply]
https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/autolinked-references-and-urls

**Reply by Moradi1212** [reply]
Think of it 
Copilot and GitHub

**Reply by Gitleyla** [reply]
ich brauche hilfe 
bitte ist jemand da der mir hilft*

**Comment by jakubMitura14** [comment]
Will it be possible to create extension that run the code check weather it compiles (or better compiles and pass test) and if not return the copilot with the prompt to correct the error?

**Reply by jookmarki** [reply]
Yes, you can create an extension that integrates with GitHub to check code for compilation errors and run tests. This extension would monitor code changes, trigger compilation checks using tools like javac, g++, etc., and execute tests (e.g., JUnit) if configured. Feedback on errors would be displayed in GitHub, potentially using AI like GPT-3 or Copilot for error correction suggestions. This integration would streamline development workflows by ensuring code quality and correctness directly within the GitHub environment.

**Comment by Askar938** [comment]


**Comment by swiftnesstutorialby** [comment]
wow!

**Comment by Shumaila787** [comment]
> ![GitHub Copilot Extensions Limited Beta announcement](https://private-user-images.githubusercontent.com/5484339/332227518-05f8a107-1e43-4846-91e1-d49cf7f7636c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc1MDQ3NzYsIm5iZiI6MTcxNzUwNDQ3NiwicGF0aCI6Ii81NDg0MzM5LzMzMjIyNzUxOC0wNWY4YTEwNy0xZTQzLTQ4NDYtOTFlMS1kNDljZjdmNzYzNmMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYwNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MDRUMTIzNDM2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MThlNDg0OGY3MDY3NmEzOWY1YzlmNjExNDRhZjM0ZTUxZjZiZDg1MmFlNjRmZmFlMzVmNDYyOTlmYmRmNzQ3MiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.NgcsC_95Hiw6-nZnWbGRvzogASipB34HJtBifcm6Zaw) GitHub Copilot Extensions is now in [Limited Public Beta](https://aka.ms/githubcopilotextensions). With extensions, you can now extend the capabilities of GitHub Copilot Chat and enhance the experience to perform a wide range of actions across third-party tools, services, and data. Create feature flags, check log errors, access API documentation, and even deploy your application to the cloud all through natural language.
> 
> Currently, access to Copilot Extensions is limited to a small set of users and we are planning to gradually roll out larger-scale access as we learn more & gather feedback. To learn more about Copilot Extensions and how to use them, check out the [documentation](https://docs.github.com/copilot/github-copilot-chat/github-copilot-extensions/about-github-copilot-extensions) and the [GitHub Marketplace](https://github.com/marketplace?type=apps&copilot_app=true).
> 
> If you're looking to build a GitHub Copilot Extension, there are several ways. Join the [Copilot Partner Program](https://github.com/features/preview/copilot-partner-program) to explore opportunities to bring your developer tools and services into the GitHub Copilot ecosystem and have extensions light up across all supported clients (VS Code, Visual Studio, and Github.com). If you're looking to build a VS Code extension specifically, documentation is available [here](https://code.visualstudio.com/api/extension-guides/chat).

Close this new

**Comment by Ramjanali100** [comment]
ramjan ali

**Comment by Ramjanali100** [comment]
ruhan

**Comment by mishmanners** [comment]
This is super awesome and can't wait to try them out :smile:

**Comment by DerAnnere** [comment]
> ![GitHub Copilot Extensions Limited Beta announcement](https://private-user-images.githubusercontent.com/5484339/332227518-05f8a107-1e43-4846-91e1-d49cf7f7636c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAxNDEyNTksIm5iZiI6MTcyMDE0MDk1OSwicGF0aCI6Ii81NDg0MzM5LzMzMjIyNzUxOC0wNWY4YTEwNy0xZTQzLTQ4NDYtOTFlMS1kNDljZjdmNzYzNmMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcwNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MDVUMDA1NTU5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTkzYTBkYmExYWIzYjI4ZWViNjQ0YmFhYTAxZDczYmJjNDEyMWFlMmE4NmMwZjhmOTIzYjg4YmJhM2NmNzJlYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.kXwWfkdxkHLB174rbn5RCjt4KX_p_HjcVgd-zwbn5Co) GitHub Copilot Extensions is now in [Limited Public Beta](https://aka.ms/githubcopilotextensions). With extensions, you can now extend the capabilities of GitHub Copilot Chat and enhance the experience to perform a wide range of actions across third-party tools, services, and data. Create feature flags, check log errors, access API documentation, and even deploy your application to the cloud all through natural language.
> 
> Currently, access to Copilot Extensions is limited to a small set of users and we are planning to gradually roll out larger-scale access as we learn more & gather feedback. To learn more about Copilot Extensions and how to use them, check out the [documentation](https://docs.github.com/copilot/github-copilot-chat/github-copilot-extensions/about-github-copilot-extensions) and the [GitHub Marketplace](https://github.com/marketplace?type=apps&copilot_app=true).
> 
> If you're looking to build a GitHub Copilot Extension, there are several ways. Join the [Copilot Partner Program](https://github.com/features/preview/copilot-partner-program) to explore opportunities to bring your developer tools and services into the GitHub Copilot ecosystem and have extensions light up across all supported clients (VS Code, Visual Studio, and Github.com). If you're looking to build a VS Code extension specifically, documentation is available [here](https://code.visualstudio.com/api/extension-guides/chat).

**Comment by israeloliveiradev** [comment]
Wow!

**Comment by PolyVaz** [comment]
Bonjour ,vous êtes gentils. Mais ça pour moi c’est très compliqué…se commence faire de choses… sur que je fait des Bêtises!!! Merci

**Comment by Tusho7** [comment]
Amazing👌🙌

**Reply by alnuaimicoder** [reply]
Good

**Comment by FreshImmuc** [comment]
Extensions seems very intuitive and great for using copilot more across workflows. Integrations such as with Teams make it easy for me to keep focus on my stuff.

**Reply by Joliverduras77** [reply]
Ok

**Comment by Glaxiq** [comment]
> ![GitHub Copilot Extensions Limited Beta announcement](https://private-user-images.githubusercontent.com/5484339/332227518-05f8a107-1e43-4846-91e1-d49cf7f7636c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkzNzk1ODMsIm5iZiI6MTcyOTM3OTI4MywicGF0aCI6Ii81NDg0MzM5LzMzMjIyNzUxOC0wNWY4YTEwNy0xZTQzLTQ4NDYtOTFlMS1kNDljZjdmNzYzNmMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMTlUMjMwODAzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZThmMGRhMDEzMmFhOTQ4MzFhMjgwMGYxMDBkNTc5NWRhODcxNGY4ZGQyODNhZmU4NWMxMTdlNWRlNjFjMjUyZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.NcBJKxVWJwN6XcwswhzvPm66HC8zc24UuEO8SRhADzg) GitHub Copilot Extensions is now in [Limited Public Beta](https://aka.ms/githubcopilotextensions). With extensions, you can now extend the capabilities of GitHub Copilot Chat and enhance the experience to perform a wide range of actions across third-party tools, services, and data. Create feature flags, check log errors, access API documentation, and even deploy your application to the cloud all through natural language.
> 
> Currently, access to Copilot Extensions is limited to a small set of users and we are planning to gradually roll out larger-scale access as we learn more & gather feedback. To learn more about Copilot Extensions and how to use them, check out the [documentation](https://docs.github.com/copilot/github-copilot-chat/github-copilot-extensions/about-github-copilot-extensions) and the [GitHub Marketplace](https://github.com/marketplace?type=apps&copilot_app=true).
> 
> If you're looking to build a GitHub Copilot Extension, there are several ways. Join the [Copilot Partner Program](https://github.com/features/preview/copilot-partner-program) to explore opportunities to bring your developer tools and services into the GitHub Copilot ecosystem and have extensions light up across all supported clients (VS Code, Visual Studio, and Github.com). If you're looking to build a VS Code extension specifically, documentation is available [here](https://code.visualstudio.com/api/extension-guides/chat).

**Comment by CodingDanny** [comment]
Is there a plan to make Copilot Extensions available within the Copilot Jetbrains Plugin in the near future?  ✌️ 
Would be highly appreciated!

**Reply by barbermjgt212** [reply]
Wow eso estaria che ere a mime cuesta entender mi copilot

**Comment by CodingDanny** [comment]
There is an Issue with the Perplexity Extension. When I try to connect, It says "This GitHub App must be configured with a callback URL", so I can not use it.
![perplexity-extension1](https://github.com/user-attachments/assets/f7d4f613-0ddb-449b-bc68-bd584c7668ad)
![perplexity-extension2](https://github.com/user-attachments/assets/d78daafe-3cc5-478a-a948-932f264018d8)

**Comment by longredzhong** [comment]
how to uninstall some github Copilot Extensions？I don't want some extensions to appear in vscode github copilot。

**Comment by whg517** [comment]
Is it possible to create an extension that can support review code in the PR phase, which can review code, do basic checks, such as spelling errors, in-depth grammar and language feature checks, API compatibility, code bugs, etc., and give some programming suggestions? Submit the results of the review as comments to the PR session.

**Comment by nsheaps** [comment]
If extensions are free, it'd be good to not ask for a payment method, feels like it may become a bait and switch if the extension ends up costing something later (and cost would be good to see in the marketplace, especially personal vs org)

Separately, it's unclear what "installing into your personal account" versus org means, especially in the context of who owns the copilot account. In addition it'd be good to allow an install of a marketplace copilot extension org-wide (eg ticketing system and use an org API key rather than each user authenticating).

**Reply by Dantheman23-coder** [reply]
great points.

**Comment by orkuhh** [comment]
It would be great if the Copilot Extensions Marketplace displayed some kind of indicator which extensions are already installed or option to hide the ones that are indeed installed. And also some kind of representations if they are free or paid so you don't have to go in and check.

**Reply by Emjayz313** [reply]
Ikr I need It

**Reply by Clarodon2** [reply]
Thank you 💕

**Reply by Mike7786267** [reply]
Great

**Comment by blockbabak** [comment]
There are so many of them now. It is hard to find one that 
1-Is useful 
2-It does what it claims to do.
I have been on a roundabout of subscribing to different "premium" extensions, agents, collaborators, whatever you want to call it.
First of all, they all have some fundamental problems like not being able to create a shared profile of the user which means I have to keep repeating the same thing for every new 'chat', 'project'.

After a while I confuse them myself. I have loads of project outlines and steps that I have had to give up on, only to find out later that something that was a dead-end is not a dead end in this other one. 

If there is one that can at least recognise my profile and allow me to set some ground rules, then I would be happy to pay for it.

If anyone knows one, then please let me know because I have arrived at the conclusion that it is all hype, and Alan Turing didn't mean just any old user being fooled. He meant someone like himself being fooled that they are talking to another person.

If they can't fool me then they definitely wouldn't fool Alan Turing.

However, I have learned a lot about the human mind that people claim to be unique or mysterious. Like how they learn to lie or being autistic. I have learned that a lot is more emergent than inherent.
